test/integration/master/kube_apiserver_test.go

```diff
@@ -190,6 +203,196 @@ func TestOpenAPIDelegationChainPlumbing(t *testing.T) {
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "apiextensions.apis.pkg.apiextensions-apiserver.k8s.io",
+			Version: "v1beta1",
+			Scope:   apiextensionsv1beta1.ClusterScoped,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:   "customresourcedefinitions",
+				Singular: "customresourcedefinition",
+				Kind:     "CustomResourceDefinition",
+				ListKind: "CustomResourceDefinitionList",
+			},
+			Validation: &apiextensionsv1beta1.CustomResourceValidation{
+				OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{
+					Type: "object",
+					Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+						testApiextensionsOverlapProbeString: {Type: "boolean"},
+					},
+				},
+			},
+		},
+	}
+	etcd.CreateTestCRDs(t, apiextensionsclient, false, crd)
+
+	// Create a probe CRD bar that triggers an OpenAPI spec change
+	if err := triggerSpecUpdateWithProbeCRD(t, apiextensionsclient, "bar"); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Expect the apiextensions definition to not change, since the overlapping definition will get renamed.
+	apiextensionsDefinition, exist, err := getOpenAPIDefinition(apiextensionsclient, `io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.CustomResourceDefinition`)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if !exist {
+		t.Fatalf("unexpected error: apiextensions definition doesn't exist")
+	}
+	bytes, err := json.Marshal(apiextensionsDefinition)
+	if exist := strings.Contains(string(bytes), testApiextensionsOverlapProbeString); exist {
+		t.Fatalf("unexpected error: apiextensions definition gets overlapped")
+	}
+}
+
+// triggerSpecUpdateWithProbeCRD creates a probe CRD with suffix in name, and waits until
+// the path and definition for the probe CRD show up in the OpenAPI spec
+func triggerSpecUpdateWithProbeCRD(t *testing.T, apiextensionsclient *apiextensionsclientset.Clientset, suffix string) error {
+	// Create a probe CRD that triggers OpenAPI spec change
+	name := fmt.Sprintf("integration-test-%s-crd", suffix)
+	kind := fmt.Sprintf("Integration-test-%s-crd", suffix)
+	group := "probe.test.com"
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{Name: name + "s." + group},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   group,
+			Version: "v1",
+			Scope:   apiextensionsv1beta1.ClusterScoped,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Plural:   name + "s",
+				Singular: name,
+				Kind:     kind,
+				ListKind: kind + "List",
+			},
+		},
+	}
+	etcd.CreateTestCRDs(t, apiextensionsclient, false, crd)
+
+	// Expect the probe CRD path to show up in the OpenAPI spec
+	// TODO(roycaihw): expose response header in rest client and utilize etag here
+	if err := wait.PollImmediate(1*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
```

Comment:
the chance is high that we only need one pass with `wait.Poll` instead of `PollImmediate`.